airflow-core/src/airflow/api_fastapi/execution_api/routes/task_instances.py

```diff
@@ -308,6 +310,15 @@ def ti_update_state(
         updated_state = ti_patch_payload.state
         query = TI.duration_expression_update(ti_patch_payload.end_date, query, session.bind)
         query = query.values(state=updated_state)
+
+        if updated_state == TerminalTIState.FAILED:
+            ti = session.get(TI, ti_id_str)
+            ser_dag = request.app.state.dag_bag.get_dag(dag_id)
+            if ser_dag and ser_dag.__dict__.get("fail_fast", False):
```

Comment:
We should really be accessing things out of the dict directly.  Does `if set_dag.fail_fast` not work?

If not then `getattr(ser_dag, "fail_fast", False)` would be better.

Reply:
`ser_dag` is of the type: `SerializedDAG` so we cannot directly do that.

But `getattr(ser_dag, "fail_fast", False)` should work, let me try that.

Reply:
Handled in 94b1fcef4453d8a5e4f0277db0c399fccb5ebc81

Reply:
Works as expected